### PR TITLE
feat: plugin to sf parity

### PR DESCRIPTION
**Pull request by peternhale**
### What does this PR do?
Moves plugin-info to sf/sfdx parity
### What issues does this PR fix or reference?
@W-12084133@

### CODE REVIEW

File: src/commands/info/releasenotes/display.ts [hunk 1]
```diff
@@ -30,30 +30,37 @@ const HIDE_FOOTER = 'SFDX_HIDE_RELEASE_NOTES_FOOTER';
...
-    hook: flags.boolean({
+    hook: Flags.boolean({
       hidden: true,
-      description: messages.getMessage('flags.hook'),
+      summary: messages.getMessage('flags.hook.summary'),
+      description: messages.getMessage('flags.hook.description'),
     }),
+    loglevel,
   };
+  private logger: Logger;
```
Comment (mshanemc) [inline src/commands/info/releasenotes/display.ts:56]: put as a const in run()

File: messages/doctor.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
+# commandDescription
```
Comment (mshanemc) [inline messages/doctor.md:1]: I love that there's a file namd doctor.md

File: src/commands/doctor.ts [hunk 3]
```diff
@@ -20,28 +20,32 @@ import { DiagnosticStatus } from '../diagnostics';
 Messages.importMessagesDirectory(__dirname);
 const messages = Messages.loadMessages('@salesforce/plugin-info', 'doctor');
 
-export default class Doctor extends SfdxCommand {
-  public static description = messages.getMessage('commandDescription');
-  public static examples = messages.getMessage('examples').split(os.EOL);
+export default class Doctor extends SfCommand<SfDoctorDiagnosis> {
+  public static readonly summary = messages.getMessage('commandDescription');
```
Comment (mshanemc) [inline src/commands/doctor.ts:24]: this command's description is crazy long.  Let's have a short summary

File: src/commands/doctor.ts [hunk 4]
```diff
@@ -53,18 +57,19 @@ export default class Doctor extends SfdxCommand {
...
   public async run(): Promise<SfDoctorDiagnosis> {
+    const { flags } = await this.parse(Doctor);
     this.doctor = SFDoctor.getInstance();
     const lifecycle = Lifecycle.getInstance();
 
-    const pluginFlag = this.flags.plugin as string;
-    const commandFlag = this.flags.command as string;
-    const outputdirFlag = this.flags.outputdir as string;
-    const createissueFlag = this.flags.createissue as boolean;
+    const pluginFlag = flags.plugin;
```
Comment (mshanemc) [inline src/commands/doctor.ts:64]: this extra consts are unnecessary now that we have flag typing (no more `as string`)

File: src/commands/doctor.ts [hunk 5]
```diff
@@ -122,9 +127,13 @@ export default class Doctor extends SfdxCommand {
         agent: { https: ProxyAgent(getProxyForUrl(raw)) },
       });
 
-      const title = await this.ux.prompt('Enter a title for your new issue');
+      const title: Record<string, string> = await this.prompt({
```
Comment (mshanemc) [inline src/commands/doctor.ts:130]: why is the typing necessary?  doesn't `prompt` default to Record<string, string>

**Review by mshanemc — APPROVED** [review]
approved with non-blocking comments

File: src/commands/doctor.ts (hunk 5 above)
Comment (peternhale) [reply src/commands/doctor.ts:130]: @mshanemc No, the prompt function returns `<R extends Prompter.Answers>` which is same as `Record<string, unknown>

**Review by WillieRuemmele — APPROVED** [review]

**Comment by WillieRuemmele** [comment]
### QA Notes
---
`Doctor`:
`sfdx doctor`: ✅ 
`-h` ✅ 
`--help`: ✅ 
generated files all match ✅ 
`-d`,`-o` interoperability ✅ 
`bin/run doctor -o out -d out1 ` -> `flags['output-dir'] = out1`
`--createissue` -> new webpage ✅ 
`--create-issue` alias ✅ 
`--plugin user` -> plugin isn't installed error ✅ 
`--plugin @salesforce/plugin-user` -> ✅ 
`--json`: ✅ 

---
`releasenotes:display` ✅ 
`whatsnew` ✅ 
`-h`/`--help` ✅ 
`-v` ✅ 
`--hook` ✅ 
`SFDX_HIDE_RELEASE_NOTES_FOOTER=true` ✅ 
`SFDX_HIDE_RELEASE_NOTES=true ../../oss/sfdx-cli/bin/run releasenotes:display -v 7.133.0 --hook` ✅ 
`--json` ✅ 

---

`sf dev audit messages` ✅

**Pull request merged**